### PR TITLE
[Feature] CON 18 - Enable disabled add criteria

### PR DESCRIPTION
**Pull request by habemuscode**
**Related to**

https://oat-sa.atlassian.net/browse/CON-18

**Related to this PR**

https://github.com/oat-sa/tao-core/pull/2719

**Description**

**as a** TAO user with access to advanced search
**I want** to be able to see and use the advanced search filters
**when** using search component

**as a** TAO user with no access to advanced search
**I want** not to see or use the advanced search filters
**when** using  search component

**AC**

1. For an environment with advanced search enabled*, ‘add criteria’ button is displayed according to mockups

2. For an environment with advanced search disabled*, ‘add criteria’ button is not displayed

*Advanced search is enabled by operations, so each AC must be tested on a different deployment OR environment variable can be updated dynamically by Luis or Andre so QA can test each AC on the same environment.

### CODE REVIEW

File: src/searchModal/advancedSearch.js [hunk 1]
```diff
@@ -131,30 +134,45 @@ export default function advancedSearchFactory(config) {
...
-        // open dropdown when user clicks on add criteria input
-        $addCriteriaInput.on('click', () => {
-            $criteriaSelect.select2('open');
-            // if dropdown is opened above addCriteria input, top property is slightly decreased to avoid overlapping with addCriteria icon
-            if ($('.criteria-dropdown-select2').hasClass('select2-drop-above')) {
-                $('.criteria-dropdown-select2').css(
-                    'top',
-                    $('.criteria-dropdown-select2').css('top').split('px')[0] - 10 + 'px'
-                );
+        const route = urlUtil.route('status', 'advanced_search', 'tao');
```
Comment (shpran) [inline src/searchModal/advancedSearch.js:137]: Route will be `tao/AdvancedSearch/status` 😄

https://github.com/oat-sa/tao-core/pull/2719/files#diff-7ef472c51d586c321f4bfb5ee093c8ad8b70f31bebf72bd2042b4a5021de0c48R27

Comment (habemuscode) [reply src/searchModal/advancedSearch.js]: Done, thanks

**Review by shpran — APPROVED** [review]
LGTM
Good job! 👍🏻 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: src/searchModal/advancedSearch.js [hunk 2]
```diff
@@ -131,30 +134,41 @@ export default function advancedSearchFactory(config) {
...
-            // if dropdown is opened above addCriteria input, top property is slightly decreased to avoid overlapping with addCriteria icon
-            if ($('.criteria-dropdown-select2').hasClass('select2-drop-above')) {
-                $('.criteria-dropdown-select2').css(
-                    'top',
-                    $('.criteria-dropdown-select2').css('top').split('px')[0] - 10 + 'px'
-                );
+        const route = urlUtil.route('status', 'AdvancedSearch', 'tao');
+        request(route).then(function (response) {
+            if (!response.enabled) {
+                $addCriteria.addClass('disabled');
```
Comment (adriloma) [inline src/searchModal/advancedSearch.js:140]: Considering that you have already added `disabled` class to `.add-criteria-container` on  `initUiSelectors`, which is executed right before `initAddCriteriaSelector`, and that this happens only once (on `render` event), I don't see when would be necessary to add `disabled` class agian.

**Review by adriloma — COMMENTED** [review]
Just minor comment. Looks good although test is missing and is an important feature to cover.

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] bumping versions are applied (if applicable)

**Review by adriloma — CHANGES_REQUESTED** [review]
Please fix failing test and consider adding new one.

File: src/searchModal/advancedSearch.js [hunk 3]
```diff
@@ -111,15 +112,19 @@ export default function advancedSearchFactory(config) {
         .setTemplate(advancedSearchTpl)
         .on('render', () => {
             initUiSelectors();
-            initAddCriteriaSelector();
-            initCriteriaState();
+            initAddCriteriaSelector().then(() => {
```
Comment (adriloma) [inline src/searchModal/advancedSearch.js:115]: You should cover `catch` exception:
```
            initAddCriteriaSelector().then(() => {
                initCriteriaState();
                instance.trigger('ready');
            }).catch(e => instance.trigger('error', e));
```

File: src/searchModal/advancedSearch.js [hunk 4]
```diff
@@ -131,30 +136,40 @@ export default function advancedSearchFactory(config) {
...
-        $addCriteriaInput.on('click', () => {
-            $criteriaSelect.select2('open');
-            // if dropdown is opened above addCriteria input, top property is slightly decreased to avoid overlapping with addCriteria icon
-            if ($('.criteria-dropdown-select2').hasClass('select2-drop-above')) {
-                $('.criteria-dropdown-select2').css(
-                    'top',
-                    $('.criteria-dropdown-select2').css('top').split('px')[0] - 10 + 'px'
-                );
+        const route = urlUtil.route('status', 'AdvancedSearch', 'tao');
+        request(route).then(function (response) {
```
Comment (adriloma) [inline src/searchModal/advancedSearch.js:140]: Same (sorry, missed this one on my previous review)

File: test/searchModal/advancedSearch/test.js [hunk 5]
```diff
@@ -98,25 +98,34 @@ define([
     });
 
     QUnit.module('advanced search logic');
-    QUnit.test('advancedSearch criteria manipulation', function (assert) {
+    QUnit. QUnit.cases.init([
```
Comment (bziondik) [inline test/searchModal/advancedSearch/test.js:101]:
```suggestion
    QUnit.cases.init([
```

**Review by adriloma — APPROVED** [review]
Please consider adding a `catch `callback to promises so the exception is captured.

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] bumping versions are applied (if applicable)

File: src/searchModal/advancedSearch.js (hunk 3 above)
Comment (habemuscode) [reply src/searchModal/advancedSearch.js:115]: done

**Review by bziondik — APPROVED** [review]
- [x]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Pull request merged**